src/EnergyPlus/OutputProcessor.cc

```diff
@@ -2955,55 +2741,52 @@ namespace OutputProcessor {
         // easier maintenance and writing of data to the SQL database.
 
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        std::string NumberOut; // Character for producing "number out"
         std::string MaxOut;    // Character for Max out string
         std::string MinOut;    // Character for Min out string
-        Real64 rmaxValue;
-        Real64 rminValue;
-        Real64 repVal; // The variable's value
         auto &rf = state.dataResultsFramework->resultsFramework;
         auto &sql = state.dataSQLiteProcedures->sqlite;
 
-        repVal = repValue;
-        if (storeType == StoreType::Averaged) {
-            repVal /= numOfItemsStored;
-        }
-        if (repValue == 0.0) {
-            NumberOut = "0.0";
-        } else {
-            NumberOut = format("{:f}", repVal);
-        }
+        Real64 repVal = (storeType == StoreType::Average) ? (StoreValue / NumStored) : StoreValue;
 
         // Append the min and max strings with date information
-        MinOut = fmt::to_string(minValue);
-        MaxOut = fmt::to_string(MaxValue);
-        ProduceMinMaxString(MinOut, minValueDate, freq);
-        ProduceMinMaxString(MaxOut, maxValueDate, freq);
-
         if (rf->timeSeriesEnabled() &&
             (freq == ReportFreq::Day || freq == ReportFreq::Month || freq == ReportFreq::Year || freq == ReportFreq::Simulation)) {
             // add to daily TS data store
-            rf->freqTSData[(int)freq].pushVariableValue(reportID, repVal);
+            rf->freqTSData[(int)freq].pushVariableValue(ReportID, repVal);
         }
 
-        rminValue = minValue;
-        rmaxValue = MaxValue;
         if (sql) {
-            sql->createSQLiteReportDataRecord(reportID, repVal, freq, rminValue, minValueDate, rmaxValue, maxValueDate);
+            sql->createSQLiteReportDataRecord(ReportID, repVal, freq, MinValue, minValueDate, MaxValue, maxValueDate);
         }
 
-        if ((freq == ReportFreq::EachCall) || (freq == ReportFreq::TimeStep) || (freq == ReportFreq::Hour)) { // -1, 0, 1
-            if (state.files.eso.good()) {
-                print(state.files.eso, "{},{}\n", reportID, NumberOut);
+        if (state.files.eso.good()) {
+            std::string NumberOut;
+            if (varType == VariableType::Real) {
```

Comment:
Ok, this is the problem. It's actually not a problem, it's just a random annoying paper-cut. Turns out that we print integer variables as if they were real (i.e., we print the decimal point and zeros). However, we also print them different than we do real variables. We print real variables using `dtoa` (which truncates the number of zeros) and we print integer variables using `format` (which does not) unless the number is 0 in which case we just print `0.0`. Why are we doing this? FWIW it doesn't really matter for anything other than unit test purposes, it's just annoying.

Reply:
For integer variables, if it's being reported at it's native `OutputProcessor::TimeStepType` then it is really an integer, and one would like to see integer values. If it's being summed, it's still an integer value. But if it's being averaged, then the reported value will be a real. My guess is that the format for integers was added to help distinguish them from reals when viewing them in a csv file, for example.